src/main/java/com/czertainly/core/service/cmp/CertificateKeyService.java

```diff
@@ -0,0 +1,54 @@
+package com.czertainly.core.service.cmp;
+
+import com.czertainly.api.clients.cryptography.CryptographicOperationsApiClient;
+import com.czertainly.api.model.common.enums.cryptography.KeyType;
+import com.czertainly.core.dao.entity.*;
+import com.czertainly.core.provider.CzertainlyProvider;
+import com.czertainly.core.provider.key.CzertainlyPrivateKey;
+import com.czertainly.core.provider.key.CzertainlyPublicKey;
+import com.czertainly.core.service.CryptographicKeyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CertificateKeyService {

```

Comment:
Do we nedd to have `CertificateKeyService` specifically for CMP implementation. It seems to me like a general service that is working with JCA/JCE CZERTAINLY Provider and should be move somewhere else, if needed.